packages/core/src/test/shared/sam/build.test.ts

```diff
@@ -551,3 +564,23 @@ describe('SAM runBuild', () => {
         })
     })
 })
+
+async function runInParallel(samLocation: SamAppLocation): Promise<[SamBuildResult, SamBuildResult]> {
+    return Promise.all([runBuild(new AppNode(samLocation)), runBuild(new AppNode(samLocation))])
```

Comment:
I worry this would cause some flakiness as the two build is run at same time. There might be a race condition that both Build run `throwIfTemplateIsBeingBuilt` both passes and then both run `registerTemplateBuild` and proceed to build. In this case the two build will be run together and cause test to fail. I would suggest we add a 0.05s wait time between them to avoid the unlikely but possible race condition.